spot-contracts/contracts/RolloverVault.sol

```diff
@@ -394,6 +400,118 @@ contract RolloverVault is
         return redemptions;
     }
 
+    /// @notice Allows users to swap their underlying tokens for perps held by the vault
+    /// @param underlyingAmtIn The amount of underlying tokens swapped in.
+    /// @param execRecover If set to true, recovers vault assets to maximize liquidity for swapping.
+    /// @return The amount of perp tokens swapped out.
+    function swapUnderlyingForPerps(uint256 underlyingAmtIn, bool execRecover)
```

Comment:
I can also imagine a design that doesn't have an `execRecover` input, but instead will lazily execute recover, if it's needed.

Is the benefit simply that it's cheaper to check execRecover from calldata/mem vs checking token balance? It does complicate the interface.

Reply:
Checking if recover is executable will require as many reads as actually executing the recover. I imagine most interfaces will invoke this function with a default value of `execRecover=false`. 

Agree we can get rid of this.